src/context/WishlistContext.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { useAuth } from "./index";
+import axios from "axios";
+
+const wishlistContext = createContext([]);
+
+const useWishlist = () => useContext(wishlistContext);
+
+function WishlistProvider({ children }) {
+  const {
+    authState: { token },
+  } = useAuth();
+  const [wishlistItems, setWishlistItems] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      if (token) {
+        try {
+          let response = await axios.get("/api/user/wishlist", {
+            headers: {
+              authorization: token,
+            },
+          });
+          if (response.status === 200) {
+            setWishlistItems(response.data.wishlist);
+          }
+        } catch (err) {
+          console.log(err);
+        }
+      }
+    })();
+  }, [token]);
+
+  const addToWishlist = async (product, setIsLoading) => {
+    try {
+      setIsLoading((prev) => ({ ...prev, wishlist: true }));
+      const response = await axios.post(
+        "/api/user/wishlist",
+        {
+          product,
+        },
```

Comment:
you can put this in one line as { product } as it doesn't contain any other element. Rest seems good to me